### PR TITLE
Allow only “others” input to go into others food category

### LINKED ISSUE #184: [PE-D] Category and Location Input Validation Flaw
For `/cat` parameter,  inputs such as  numbers, symbols, negative numbers, Non-English texts will be categorized as `OTHERS`.
This can be seen for `/loc` parameter as well.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: tzexern/ped#10

### PR DESCRIPTION
**Pull request by Leeyp**
Closes #184

### CODE REVIEW

**Review by Vinci-Hu — APPROVED** [review]
LGTM. Also reminds me of the need to modify docs. 💯

**Review by SimJJ96 — APPROVED** [review]
LGTM

**Comment by kwokyto** [comment]
> LGTM. Also reminds me of the need to modify docs. 💯

no worries, i can do this for the UG @Vinci-Hu

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -143,9 +143,9 @@ created for the FridgeFriend project, and stored inside the `Exceptions` compone
 All exceptions extend from the Java default `Exception` class. They only differ with regard
 to the throwable error message.
 
-The `Exceptions` component currently consists of nine (9) custom Exceptions, as of v2.0:
+The `Exceptions` component currently consists of eleven 11 custom Exceptions, as of v2.0:
```
Comment (kwokyto) [inline docs/DeveloperGuide.md]: do you want to put brackets around 11?

File: src/main/java/seedu/fridgefriend/exception/InvalidFoodLocationException.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package seedu.fridgefriend.exception;
+
+//@@author leeyp
+/**
+ * Signals that the food category does not match any existing category.
+ */
+public class InvalidFoodLocationException extends Exception {
+
+    private static final String messagePart1 = "Sorry my friend, ''";
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/exception/InvalidFoodLocationException.java:9]: can we change this HAHA we have a lot of "Sorry my friend"s
maybe like Sorry amigo or smth

File: src/main/java/seedu/fridgefriend/utilities/Parser.java [hunk 3]
```diff
@@ -357,7 +358,7 @@ private static Command getClearCommand() {
         return clearCommand;
     }
 
-    //@@author kwokyto
+    //@@author leeyp
     /**
      * Returns a HistoryCommand object.
      *
```
Comment (kwokyto) [inline src/main/java/seedu/fridgefriend/utilities/Parser.java:368]: line 365 should return HistoryCommand instead

**Review by kwokyto — APPROVED** [review]
Otherwise, LGTM

**Pull request merged**